src-extras/Database/LSMTree/Extras/Generators.hs

```diff
@@ -454,8 +457,31 @@ packRawBytesPinnedOrUnpinned True  = \ws ->
       return mba
 
 shrinkRawBytes :: RawBytes -> [RawBytes]
-shrinkRawBytes (RawBytes pvec) = [ RawBytes (VP.fromList ws)
-                                 | ws <- QC.shrink (VP.toList pvec) ]
+shrinkRawBytes (RawBytes pvec) =
+    [ RawBytes pvec'
+    | pvec' <- shrinkVec (take 1 . QC.shrink) pvec
+                          -- no need to try harder shrinking individual bytes
```

Comment:
My first thought is that I disagree. It would be nice to make `prop_forAllArbitraryAndShrinkPreserveInvariant` faster, but I'd argue getting good shrinks should be the goal, and arguably you shouldn't restrict the shrink space too much to do that. Maybe you could give an idea about why `take 1 . QC.shrink` is sufficient in your opinion?

Reply:
Arguably shrinking the bytes of serialised keys is more important than shrinking the bytes of serialised values. Maybe we should have different shrinkers for the two?

Reply:
We now try up to two shrinks (0 and ``x `div` 2``) to at least allow getting to smaller numbers via iterative shrinking. Also, if we really encounter a test case where the shrinking doesn't seem ideal, we can still go back to a more exhaustive one and re-run the test.